addons/web/static/src/core/dialog/dialog.js

```diff
@@ -90,7 +90,14 @@ export class Dialog extends Component {
     }
 
     onEscape() {
-        this.data.close();
+        return this.dismiss();
+    }
+
+    async dismiss() {
+        if (this.data.dismiss) {
+            await this.data.dismiss();
```

Comment:
I'm wondering if we should protect this from being called twice (e.g. press escape twice quickly)